### DISCUSSION TITLE
评级列有必要做吗？

### DISCUSSION THREAD

**Discussion by MuiseDestiny** [MuiseDestiny/zotero-style#142] [Announcements]
![image](https://user-images.githubusercontent.com/51939531/219393315-5adbf2fc-63e3-4f0e-aa8c-4f016f222816.png)

Zotero本身没有评级功能，可能长时间以来我们都发展了一套自己习惯的文献标记方法。此时类似Endnote的评级功能是否还有必要做？

已知Zotero IF高级版有这个功能，但只是在文献右键时候出现，style如果做，将会做成Endnote那样，点击一次即可完成/修改评级。
![image](https://user-images.githubusercontent.com/51939531/219393829-53bb266c-ff81-48ab-a3be-006614e9eb6e.png)

**Comment by yfzhao20** [comment]
其实配合颜色标签可以实现，但是如果有单独一列的优先级，那还是挺不错的，多了一种排序方式。在 NoteExpress 上也用过类似的优先级

![图片](https://user-images.githubusercontent.com/80528249/219548713-210e7ea7-0532-4784-89bf-cd237e19402b.png)

**Comment by MuiseDestiny** [comment]
![image](https://user-images.githubusercontent.com/51939531/219626595-1cdd998e-18d4-47e5-ae7d-670ce0f68ef1.png)

![018744867da0c3db9a9150816b14941](https://user-images.githubusercontent.com/51939531/219626678-23d11e31-b887-4bb4-800a-9657889b4dca.png)

可以用任何你喜欢的emoj

**Reply by yfzhao20** [reply]
用不同颜色表示优先级或许也是一种方法？想法来自 NE

![图片](https://user-images.githubusercontent.com/80528249/219645286-c5c7c8a0-84e9-4224-87ec-926ed13602c3.png)

感觉虽然不如星星多少直观，但是比较紧凑。

> 或者干脆在列设置里开自定义？

**Reply by MuiseDestiny** [reply]
但是这样不能一步点击操作了，需要二级菜单，这个可以通过标签来实现的。

**Comment by xplusyi** [comment]
好像没找到赞赏通道:-)

**Comment by Wenwen-ai-cell** [comment]
先说：很感谢作者大大开发这么多功能，真的很实用！！！
今天刚吐槽了评级列表，出现bug了，想要3颗星，必须点到第四个星星处，这样第五个星星就点不开，此外，评级列表很容易误触，而且颜色不直观，感觉作用不大

**Reply by MuiseDestiny** [reply]
对，我很少用它，一般用标签了。而且zotero阻止了顶层div向内的事件传播，让类似评级这种互动很难实现。

**Reply by zhuyuhua23** [reply]
评级怎么弄啊？需要啊。怎么改符号啊？mark和option设置是啥区别？

**Reply by zhuyuhua23** [reply]
> ![image](https://user-images.githubusercontent.com/51939531/219393315-5adbf2fc-63e3-4f0e-aa8c-4f016f222816.png)
> 
> Zotero本身没有评级功能，可能长时间以来我们都发展了一套自己习惯的文献标记方法。此时类似Endnote的评级功能是否还有必要做？
> 
> 已知Zotero IF高级版有这个功能，但只是在文献右键时候出现，style如果做，将会做成Endnote那样，点击一次即可完成/修改评级。 ![image](https://user-images.githubusercontent.com/51939531/219393829-53bb266c-ff81-48ab-a3be-006614e9eb6e.png)

评级怎么弄啊？需要啊。怎么改符号啊？mark和option设置是啥区别？

**Reply by MuiseDestiny** [reply]
看下endnote
是这样子的：⭐⭐⭐🔘🔘

⭐是mark
🔘是option

**Comment by gaochuanchao** [comment]
我还是挺喜欢这个功能的, 在复习一些paper的时候能够比较快的找到那些重要的paper. 非常感谢作者的付出.

**Comment by zhuyuhua23** [comment]
评级怎么弄啊？需要啊。怎么改符号啊？mark和option设置是啥区别？

**Comment by 11good11** [comment]
其实我非常喜欢这个评级功能，我一般用评级功能对文章重要程度作标记，而颜色标注对研究领域作标注，非常建议保留这个功能，有些人如果不习惯这个功能话，完全可以不把这个功能调出来，如果盲目地去去掉话，那些已经习惯这功能的用户反而会不适用

**Comment by zhuyuhua23** [comment]
&nbsp;评级列为啥取消了啊？很有必要啊！求恢复，谢谢啊！


朱昱桦(Yuhua&nbsp;Zhu)
***@***.***

贵州省地质调查院
Guizhou&nbsp;Geological&nbsp;Survey


&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "MuiseDestiny/zotero-style"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年6月14日(星期三) 晚上7:40
***@***.***&gt;;
抄送:&nbsp;"朱昱桦(Yuhua ***@***.******@***.***&gt;;
主题:&nbsp;Re: [MuiseDestiny/zotero-style] 评级列有必要做吗？ (Discussion #142)





 
看下endnote
 是这样子的：⭐⭐⭐🔘🔘
 
⭐是mark
 🔘是option

**Comment by JinGao-psych** [comment]
请问评级部分显示乱码怎么办呀？没有星星或者圆圈。

此外，标签更改了高亮的颜色之后，文献被高亮之后想删除此前的高亮部分也会报错：error：openAnnotationPopup

**Comment by xp-one** [comment]
> ![image](https://user-images.githubusercontent.com/51939531/219393315-5adbf2fc-63e3-4f0e-aa8c-4f016f222816.png)
> 
> Zotero本身没有评级功能，可能长时间以来我们都发展了一套自己习惯的文献标记方法。此时类似Endnote的评级功能是否还有必要做？
> 
> 已知Zotero IF高级版有这个功能，但只是在文献右键时候出现，style如果做，将会做成Endnote那样，点击一次即可完成/修改评级。 ![image](https://user-images.githubusercontent.com/51939531/219393829-53bb266c-ff81-48ab-a3be-006614e9eb6e.png)

很好用很直观啊，是什么原因让你产生了怀疑，每个人可以根据自己的情况直观快速的标记文章。  还有顺带说下，style需要一个切换视图的快捷键，这应该很实用的！

**Comment by xuanera33** [comment]
我是直接找了个不常用的条目手动添加🌟，比如我找的Rights，几颗星都行。每次添还可以只用emoji输入一颗，然后从下拉菜单里选几颗。很方便，平常也可以选择显示这一列，就跟作者制作这个ranking的效果一样，但我这样可以最后输出笔记时和其他信息一起显示。
![Screenshot 2024-02-01 at 11 58 46 AM](https://github.com/MuiseDestiny/zotero-style/assets/96938835/e355bb0b-f021-4176-804a-8257ff1b2a88)

![Screenshot 2024-02-01 at 11 59 36 AM](https://github.com/MuiseDestiny/zotero-style/assets/96938835/9d4224b5-7ad1-4005-894a-beae027b9916)
![Screenshot 2024-02-01 at 12 01 25 PM](https://github.com/MuiseDestiny/zotero-style/assets/96938835/3b7550b8-a079-45b9-9da4-a5ae734f5936)

**Comment by xiaoqiangch** [comment]
可以做的，但是不要用星星，直接用数字显示级别好了，列宽已经很紧张了，5颗星星占了太多空间了

**Comment by zhu-zhiyun** [comment]
评级功能还是很有必要的。新版“评级”功能出现了Bug，变成了标签，并在“标题”文字前显示。希望作者能够重视并修改此Bug。https://github.com/MuiseDestiny/zotero-style/discussions/1072

**Comment by LynnXie00** [comment]
我感觉实在是非常取决于个人习惯。我实际上确实会把文献质量评级的，但是我只分三种类别：很重要的很好的，经典文献，普通文章。所以我事实上并不用评级，而是用emoji标签的tag.... 重要就打星星，经典就打心
我觉得读文献已经够累了，在实际应用过程中简化应用方法挺好的2333
![image](https://github.com/user-attachments/assets/b9102da8-e758-4ae2-b50a-33dc33e1601f)
但是如果有重度依赖评级列来实现类似功能的人，一下子取消这个功能可能带来重大损失……

**Comment by shikily** [comment]
个人感觉不是很必要.
首先,感谢付出🙏.其次,个人的阅读习惯是会给文献评级,所以“评级”这个功能是必要的.但是,在评级列这个功能出来之前,本人已经用zotero自带的颜色标签给文献评级了,操作起来也很方便.所以,这个功能貌似(对我来说)并不是不可替代的/有独特性.从简单有效的原则来看,它是冗余的.

<img width="320" alt="截屏2024-10-30 11 18 55" src="https://github.com/user-attachments/assets/fd5e09c4-ea18-4fa5-964a-6fdaf333fb37">

**Comment by Physics-Lee** [comment]
有必要
